internal/provider/provider_test.go

```diff
@@ -73,3 +79,54 @@ func TestDataSourceSchemas(t *testing.T) {
 		})
 	}
 }
+
+func validateDocumentation(name string, resp *resource.SchemaResponse) {
+	checkDescriptor(name, resp.Schema, &resp.Diagnostics)
+	validateAttributes(name, resp.Schema.GetAttributes(), &resp.Diagnostics)
+	validateBlocks(name, resp.Schema.GetBlocks(), &resp.Diagnostics)
+}
+
+func validateDSDocumentation(name string, resp *datasource.SchemaResponse) {
+	checkDescriptor(name, resp.Schema, &resp.Diagnostics)
+	validateAttributes(name, resp.Schema.GetAttributes(), &resp.Diagnostics)
+	validateBlocks(name, resp.Schema.GetBlocks(), &resp.Diagnostics)
+}
+
+func validateAttribute(name string, attr schema.Attribute, diagnostics *diag.Diagnostics) {
+	checkDescriptor(name, attr, diagnostics)
+	if nested, ok := attr.(schema.NestedAttribute); ok {
+		validateAttributes(name, nested.GetNestedObject().GetAttributes(), diagnostics)
+	}
+}
+
+func validateBlock(name string, block schema.Block, diagnostics *diag.Diagnostics) {
+	checkDescriptor(name, block, diagnostics)
+	validateAttributes(name, block.GetNestedObject().GetAttributes(), diagnostics)
+	validateBlocks(name, block.GetNestedObject().GetBlocks(), diagnostics)
+}
+
+func validateAttributes[T schema.Attribute](name string, attrs map[string]T, diagnostics *diag.Diagnostics) {
+	for k, v := range attrs {
+		validateAttribute(name+"."+k, v, diagnostics)
+	}
+}
+
+func validateBlocks[T schema.Block](name string, blocks map[string]T, diagnostics *diag.Diagnostics) {
+	for k, v := range blocks {
+		validateBlock(name+"."+k, v, diagnostics)
+	}
+}
+
+type descriptor interface {
+	GetDescription() string
+	GetMarkdownDescription() string
+}
+
+func checkDescriptor(name string, d descriptor, diagnostics *diag.Diagnostics) {
```

Comment:
Nice!

Reply:
> Almost net negative PR 😅 +165 −143
it'll be negative in the future when more schemas are added ;-)